pandas/core/groupby/ops.py

```diff
@@ -616,8 +616,8 @@ def agg_series(self, obj: Series, func):
             # TODO: can we get a performant workaround for EAs backed by ndarray?
             return self._aggregate_series_pure_python(obj, func)
 
-        elif isinstance(obj.index, MultiIndex):
-            # MultiIndex; Pre-empt TypeError in _aggregate_series_fast
+        elif obj.index._has_complex_internals:
```

Comment:
This now excludes `PeriodIndex`, which previously worked fine since `.values` converted to a numpy array.  It looks more performant to exclude `PeriodIndex` though, since we avoid the conversion to numpy:
```python
In [1]: import numpy as np 
   ...: import pandas as pd 
   ...: from string import ascii_letters 
   ...:  
   ...: np.random.seed(123) 
   ...: group = np.random.choice(list(ascii_letters), 10**5) 
   ...: value = np.random.randint(12345, size=10**5) 
   ...: index = pd.period_range("2000", freq="D", periods=10**5) 
   ...: df = pd.DataFrame({"group": group, "value": value}, index=index)

In [2]: %timeit df.groupby("group").agg({"value": pd.Series.nunique})
17.8 ms ± 48.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  # on this branch
95.9 ms ± 183 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # on master
```